react-app-drawer/my-app/src/components/AppDrawer.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+import { Link, Outlet } from 'react-router-dom';
+import { FaBars } from 'react-icons/fa';
+
+export function Foo() {
+  return <FaBars />;
+}
+
+export type MenuItem = {
+  name: string;
+  iconUrl: string;
+  path: string;
+};
+
+type Props = {
+  menuItems: MenuItem[];
+};
+export function AppDrawer({ menuItems }: Props) {
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <div className="flex">
+      {/* Sidebar */}
+
+      <nav className="px-4 text-white bg-gray-900">
+        <button onClick={() => setIsOpen(!isOpen)}>{isOpen ? '<' : '>'}</button>
+        <ul>
```

Comment:
You should conditionally render the app title here (only if the drawer is open).